agh_graphs/production.py

```diff
@@ -0,0 +1,41 @@
+"""
+This module contains the basic code for productions. You can add your own
+production by extending the `Production` class.
+"""
+from abc import ABC, abstractmethod
+from enum import Enum
+
+from networkx import Graph
+
+
+class ProductionType(Enum):
+    layer_creation = 1
+    layer_modification = 2
```

Comment:
I quite don't get why do we need this. Can't we just have `Production` and execute them in order given by user? As far as i understand in `Evaluator` we execute `creation` productions at first and then `modification` ones but can't get why.

Reply:
It evaluates faster, by dividing the poductions into two groups we can speed up evaluating by not applying unnecessary productions.